Firestore/core/src/firebase/firestore/local/local_serializer.cc

```diff
@@ -148,6 +152,23 @@ google_firestore_v1_Document LocalSerializer::EncodeDocument(
   return result;
 }
 
+Document LocalSerializer::DecodeDocument(
+    Reader* reader,
+    const google_firestore_v1_Document& proto,
+    bool has_committed_mutations) const {
+  FieldValue::Map fields_internal =
+      rpc_serializer_.DecodeFields(reader, proto.fields_count, proto.fields);
+  SnapshotVersion version =
+      rpc_serializer_.DecodeVersion(reader, proto.update_time);
+
+  DocumentState state = has_committed_mutations
+                            ? DocumentState::kCommittedMutations
+                            : DocumentState::kSynced;
+  return Document(ObjectValue::FromMap(std::move(fields_internal)),
```

Comment:
Hmm, perhaps `DecodeFields` should return an `ObjectValue` instead, like `FSTSerializerBeta` does?